### PR TITLE
Externalize SMBLoris

### PR DESCRIPTION
**Pull request by acammack-r7**
Converts the smbloris denial of service module into an external module to help reliability and performance. Adds some Ruby-specific external module tooling as a result.

Also, the module is made less verbose and deals more gracefully with limited file descriptors.

## Verification
- [x] The steps in #8796 still work

### CODE REVIEW

**Comment by wvu-r7** [comment]
:bowing_man:

File: modules/auxiliary/dos/smb/smb_loris.rb [hunk 1]
```diff
@@ -1,89 +1,95 @@
...
+          Metasploit.log "#{sockets.length} socket(s) open", level: 'info'
         end
+      elsif n_loops % 1000 == 0
+        Metasploit.log "Holding steady at #{sockets.length} socket(s) open", level: 'info'
       end
+    rescue Interrupt
+      break
+      sockets.each &:close
+    rescue Errno::EMFILE
+      Metasploit.log "At open socket limit with #{sockets.length} sockets open. Try increasing you system limits.", level: 'warning' unless warned
```
Comment (wvu-r7) [inline modules/auxiliary/dos/smb/smb_loris.rb]: > Try rerouting auxiliary power to your system limits.

?

Comment (acammack-r7) [reply modules/auxiliary/dos/smb/smb_loris.rb]: Yeah, I guess we should direct the user to the module docs and provide help on raising the limits on various systems.

Comment (sempervictus) [reply modules/auxiliary/dos/smb/smb_loris.rb]: Potential rabbit hole with all the containers there, but yeah, maybe even some scriptage generation to set em for popular OS'.

**Review by sempervictus — COMMENTED** [review]
Thanks as always sir.
*sigh* I need to read thru the bridge code better, but the whole notion of not using internal gearing makes me a bit uneasy. That said, I do like the idea of dropping things elsewhere to run with central coordination, though that seems more a mettle function :-). 
Code looks sane, will merge test.

**Comment by busterb** [comment]
Nice to see an external ruby module. Isolation is nice, especially for resource-hungry modules like this one.

**Comment by busterb** [comment]
## Release Notes

This makes the SMBLoris module into an external 'coldstone' module. This improves reliability and performance while executing this module, while providing new infrastructure for running external Ruby modules.

**Pull request merged**